src/social_auth/backends/__init__.py

```diff
@@ -722,35 +717,30 @@ def get_backends(force_load=False):
     A force_load boolean arg is also provided so that get_backend
     below can retry a requested backend that may not yet be discovered.
     """
-    if not BACKENDSCACHE or force_load:
-        for auth_backend in setting('AUTHENTICATION_BACKENDS'):
-            mod, cls_name = auth_backend.rsplit('.', 1)
-            module = import_module(mod)
-            backend = getattr(module, cls_name)
-
-            if issubclass(backend, SocialAuthBackend):
-                name = backend.name
-                backends = getattr(module, 'BACKENDS', {})
-                if name in backends and backends[name].enabled():
-                    BACKENDSCACHE[name] = backends[name]
+    global BACKENDSCACHE
+
+    with _import_lock:
+        if not BACKENDSCACHE or force_load:
```

Comment:
Can we move the `BACKENDSCACHE` check outside the lock:

```python
if not BACKENDSCACHE or force_load:
    with _import_lock:
        ...
```

This way we only acquire the lock once while populating the cache.